procfile/procfile.go

```diff
@@ -57,3 +65,58 @@ func NewProcfileFromPath(path string) (Procfile, error) {
 
 	return p, nil
 }
+
+// NewProcfileFromBinding creates a Procfile by reading Procfile from bindings if it exists.  If it does not exist, returns an
+// empty Procfile.
+func NewProcfileFromBinding(binds libcnb.Bindings) (Procfile, error) {
+
+	p := Procfile{}
+	if binding, ok, err := bindings.ResolveOne(binds, bindings.OfType(BindingType)); err != nil {
+		return nil, fmt.Errorf("unable to resolve binding\n%w", err)
+	} else if ok {
+		if path, ok := binding.SecretFilePath(BindingType); ok {
+			if p, err = NewProcfileFromPath(filepath.Dir(path)); err != nil {
+				return nil, err
+			}
+			return p, nil
+		} else {
+			return nil, fmt.Errorf("unable to find Procfile from binding")
+		}
+	} else {
+		return p, nil
+	}
+}
+
+// NewProcfileFromPathOrBinding attempts to create a merged Procfile from given path and bindings.  If neither can be created, returns an
+// empty Procfile.
+func NewProcfileFromPathOrBinding(path string, binds libcnb.Bindings) (Procfile, error) {
+
+	procPath, procBind := Procfile{}, Procfile{}
+	var pathErr, bindErr error
+
+	if procPath, pathErr = NewProcfileFromPath(path); pathErr != nil {
+		return nil, pathErr
+	}
+	if procBind, bindErr = NewProcfileFromBinding(binds); bindErr != nil {
+		return nil, fmt.Errorf("unable to parse Procfile from binding %s\n%w", procBind, bindErr)
+	}
+	if len(procBind) > 0 && len(procPath) > 0 {
```

Comment:
You might be able to reduce this as also. I don't think that your merge function is going to care if the Procfiles are empty. It's going to iterate over the list of Procfiles passed in, and if one is empty the inner loop is just going to do nothing.

Maybe worth testing a bit, but another possible way you could clean up this method a bit.

Reply:
Yes it works like this, much better thanks 👍